src/Analyser/TypeSpecifier.php

```diff
@@ -1303,6 +1303,26 @@ private function specifyTypesForConstantStringBinaryExpression(
 			)->setRootExpr($rootExpr);
 		}
 
+		if (
+			$context->false()
+			&& $exprNode instanceof FuncCall
+			&& $exprNode->name instanceof Name
+			&& in_array(strtolower((string) $exprNode->name), ['trim', 'ltrim', 'rtrim'], true)
+			&& isset($exprNode->getArgs()[0])
+			&& $constantStringValue === ''
+		) {
+			$argType = $scope->getType($exprNode->getArgs()[0]->value);
+
+			if ($argType->isString()->yes()) {
```

Comment:
I'd say that if you omit `if ($argType->isString()->yes()) {` and do `new IntersectionType([new StringType(), new AccessoryNonEmptyStringType()])`, we could also narrow `mixed` to `non-empty-string`.